### ISSUE TITLE
Doesn't compile in release mode

### ISSUE THREAD

**Issue by cdybdahl-supwiz** [SyedAhkam/sass-rocket-fairing#1]
I got an error when compiling my project in release mode:

```
   Compiling sass-rocket-fairing v0.1.0
error[E0599]: no method named `compile_all_and_write` found for reference `&ContextManager` in the current scope
   --> /home/.../.cargo/registry/src/github.com-1ecc6299db9ec823/sass-rocket-fairing-0.1.0/src/lib.rs:108:21
    |
108 |         ctx_manager.compile_all_and_write();
    |                     ^^^^^^^^^^^^^^^^^^^^^ method not found in `&ContextManager`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `sass-rocket-fairing` due to previous error
```

It seems like it would be due to this config:

https://github.com/SyedAhkam/sass-rocket-fairing/blob/d1acd56d4641ec20681b0799e0e37e69232bd2cf/src/context.rs#L58

**Comment by sedbytes** [comment]
I have pushed a change that checks for debug mode and acts accordingly.

Try again using the master branch and update if it works as expected now.

**Comment by Mtihc** [comment]
I just ran into this issue (in july 2022).

Indeed, using the master branch works. I put this in my `Cargo.toml`

```toml
[dependencies.sass-rocket-fairing]
version = "0.1"
git = "https://github.com/SyedAhkam/sass-rocket-fairing.git"
```

Any prospects of releasing 0.1.1 or 0.2.0 or whatever?

**Comment by sedbytes** [comment]
Just released v0.2.0 🚀 (and yanked the old release)

One breaking change was introduced:

```rs
rocket::build().attach(SassFairing())
```

is now

```rs
rocket::build().attach(SassFairing::default())
```

**Comment by Mtihc** [comment]
Thank you for your quick response and action!

But..........

It seems like compilation of `.sass` files stopped working in version 0.2.0.

I'm getting this error:

```
✨ Sass:
   >> sass directory: static/sass
   >> css directory: static/css
   >> pre-compiling sass files
Error: Failed to compile file '_vars.sass'
Error: Sass error: "Error: Unknown source format: \"_vars.sass\""
```

My `_vars.sass` file is really simple. So, I assume that's not the problem.

I just noticed you added [this](https://github.com/SyedAhkam/sass-rocket-fairing/commit/d2f16a047f2f2e3037f33f99308b16eafbee7e0b#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R82) to the README

```diff
+ ## Todo
+
+ - [ ] Add support for sass (sass != scss) syntax.
+ - [ ] Combine multiple sass files into one css file.

```

I'm confused. This crate is called `sass-rocket-fairing`. Are you aware that `sass` compilation actually worked in v0.1.0 ? And now it doesn't? Since you don't mention it as a breaking change...
